### ISSUE TITLE
WebDriver not able to navigate to the parent frame

### ISSUE THREAD

**Issue by teyc** [ariya/phantomjs#13647]
TLDR: there's a bug in GhostDriver that is breaking `session/f3f7de70-720c-11e5-b529-bfb0583955f4/frame/parent`

I hit this error yesterday:

**An unhandled exception of type 'OpenQA.Selenium.WebDriverException' occurred in WebDriver.dll**
**Additional information: Unexpected error. Invalid Command Method**

And the stack trace (cleaned up showed this)

```
 {"headers":
  {"Accept":"application/json, image/png","Connection":"Close","Content-Length":"0","Content-Type":"application/json;charset=utf-8","Host":"localhost:51794"},"httpVersion":"1.1","method":"POST","post":"",
   "url":"/frame/parent",
   "urlParsed":
   {"anchor":"", "query":"",
    "file":"parent",
    "directory":"/frame/",
    "path":"/frame/parent",
    "relative":"/frame/parent",
    "port":"", "host":"", "password":"","user":"",
    "userInfo":"","authority":"","protocol":"",
    "source":"/frame/parent",
    "queryKey":{},
    "chunks":["frame","parent"]
    },
 "urlOriginal":"/session/f3f7de70-720c-11e5-b529-bfb0583955f4/frame/parent"
 }
```

The trace back shows 

```
"line":202,
"sourceURL":":/ghostdriver/request_handlers/session_request_handler.js",
 "stack":"
   at _handle (:/ghostdriver/request_handlers/session_request_handler.js:202:66)
   at _reroute (:/ghostdriver/request_handlers/request_handler.js:61:20)
   at _handle (:/ghostdriver/request_handlers/router_request_handler.js:78:46)"
```

Reading [session_request_handler.js](https://github.com/ariya/phantomjs/blob/master/src/ghostdriver/request_handlers/session_request_handler.js#L202) it appears that [postFrameCommand](https://github.com/ariya/phantomjs/blob/master/src/ghostdriver/request_handlers/session_request_handler.js#L157) wasn't dispatched because `urlParsed.file` is `parent` not `frame`.

It appears that a `FRAME_DIR` style directive should be added to the list of constants in order for this to work. Unfortunately, I'm having difficulty building PhantomJS on windows, and wonder if anyone can give me a hand?

**Comment by zackw** [comment]
Is `session/f3f7de70-720c-11e5-b529-bfb0583955f4/frame/parent` part of the PhantomJS distribution?  If not, please construct a minimized test case that you can share with us.

**Comment by teyc** [comment]
Hi Zack! I've posted the full URL captured at the exception for reference.I
think I've a fix now, but I'm not sure about how to write a test case. I'll
submit a PR over the next week.

**Comment by bertyhell** [comment]
+1

I ran into this issue as well.
When executing this code:
```csharp
driver.SwitchTo().ParentFrame().SwitchTo().Frame(1);
```

Is there a workaround possible?

```json
{
    "headers": {
        "Accept": "application/json, image/png",
        "Connection": "Close",
        "Content-Length": "2",
        "Content-Type": "application/json;charset=utf-8",
        "Host": "localhost:55687"
    },
    "httpVersion": "1.1",
    "method": "POST",
    "post": "{}",
    "url": "/frame/parent",
    "urlParsed": {
        "anchor": "",
        "query": "",
        "file": "parent",
        "directory": "/frame/",
        "path": "/frame/parent",
        "relative": "/frame/parent",
        "port": "",
        "host": "",
        "password": "",
        "user": "",
        "userInfo": "",
        "authority": "",
        "protocol": "",
        "source": "/frame/parent",
        "queryKey": {},
        "chunks": [
            "frame",
            "parent"
        ]
    },
    "urlOriginal": "/session/2e156bd0-915d-11e5-99ab-19eea4c7c838/frame/parent"
}
```

**Comment by teyc** [comment]
Yes there is a work around. I don't have the api reference on hand but you can switch to the main frame/ top level frame instead of switching to the parent frame.

-----Original Message-----

**Comment by bertyhell** [comment]
Sweet,  you're a life saver

**Comment by stiemannkj1** [comment]
Here's the workaround I use:

    driver.switchTo().window(driver.getWindowHandle());
